lib/vintage_net_mobile/modem/automatic/discovery.ex

```diff
@@ -0,0 +1,58 @@
+defmodule VintageNetMobile.Modem.Automatic.Discovery do
+  @moduledoc false
+
+  # A process for trying to discover a the type of modem and some basic
+  # information to allow dynamic configuration items
+
+  use GenServer
+
+  alias Circuits.UART
+
+  @quectel_vender_id 0x2C7C
+
+  @modems %{
+    {@quectel_vender_id, 0x0125} => VintageNetMobile.Modem.QuectelEC25,
+    {@quectel_vender_id, 0x0296} => VintageNetMobile.Modem.QuectelBG96
+  }
+
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  @impl GenServer
+  def init(args) do
+    modem = detect_modem()
+
+    {:ok, %{modem: modem, raw_config: args[:raw_config]}, {:continue, :configure_vintage_net}}
+  end
+
+  @impl GenServer
+  def handle_continue(:configure_vintage_net, state) do
+    config =
+      put_in(state, [:raw_config, :source_config, :vintage_net_mobile, :modem], state.modem)
+
+    VintageNet.configure(state.raw_config.ifname, %{type: VintageNet, vintage_net_mobile: config},
+      persist: false
```

Comment:
I think this is fine, but more just curious what the motivation _not_ to persist this is?

Reply:
The reason not to persist is how do we know the configuration is supposed to be automatic vs a configuration that was set by the config.

For example, when switching modems around (which might have not happened often to be fair) if we persist we don't know if we should invalidate the current config or not. If we never persist then we always try to resolve the modem. This might make it take longer to get the interface configured and up, but from my experience in development, it is not really noticeable.